ludwig/data/preprocessing.py

```diff
@@ -1661,6 +1669,62 @@ def _handle_missing_values(
         raise ValueError(f"Invalid missing value strategy {missing_value_strategy}")
 
 
+def handle_llm_prompt_injection(
+    dataset_cols, feature_configs, feature_name_to_preprocessing_parameters, backend
+) -> None:
+    """If output feature is a category feature and it's preprocessing has prompt_template, then we need to inject
+    the prompt into the input text feature(s)."""
+
+    # Recover input and output features
+    input_features = []
+    output_features = []
+    for feature in feature_configs:
+        if ENCODER in feature:
+            input_features.append(feature)
+        elif DECODER in feature:
+            output_features.append(feature)
+
+    # If output feature is a category feature and it's preprocessing has prompt_template, then we need to
+    # inject the prompt into the input text feature(s). Currently assumes only one output feature.
+    feature_injection_feature_names = None
+    labels = None
+    prompt = None
+    for output_feature in output_features:
+        if output_feature[TYPE] != CATEGORY:
+            continue
+        if "prompt_template" not in feature_name_to_preprocessing_parameters[output_feature[COLUMN]]:
```

Comment:
may be wrong, but are you sure that the right key here and in the following lines is column instead of name or processed column?

Reply:
Column seems to work, but I think `name` is safer so I updated this